src/db/notes.py

```diff
@@ -25,7 +38,51 @@ def create_note(
         is_public=is_public,
         author_id=author_id,
     )
-    db.add(db_note)
-    db.commit()
-    db.refresh(db_note)
-    return db_note
+    with get_db() as db:
+        db.add(db_note)
+        db.commit()
+        db.refresh(db_note)
+        return db_note
+
+
+def update_note(
+    note_id: int,
+    note_title: str,
+    note_text: str,
+    author_id: int,
+    is_public: bool = False,
+) -> Note | None:
+    with get_db() as db:
```

Comment:
If the user only wants to update the title, then potentially they don't include the note_text and is_public, so those default in app.py to None and False, respectively. The default values then get passed all the way to the NoteDB, where they aren't checked and will update the note, which would unintentially change fields that the user didn't intend.

Reply:
Right now, the only API we have for updating is a PUT request (normally used for updated/replacing a whole object).

I understand what you're saying from a DAO layer though. I will updates this to update only provided fields and that way we can have something like a PATCH request in the future.